File: _static/archives/2010/mcpc2010/voting/voting.java
// Voting, MCPC 2010 Problem C
//   Andy Harrington

import java.util.*;
import java.io.*;

public class voting {
    public static void main(String[] args) throws Exception {
        Scanner in = new Scanner(new File("voting.in"));
        String s = in.nextLine();
        while (!s.equals("#")) {
            if (s.length() < 2 || s.length()>70)  //judge check 
                 System.out.println("LONG");
            int y = 0, n = 0, p = 0, a = 0;
            for (int i = 0; i < s.length(); i++) {
                char ch = s.charAt(i);
                if (ch == 'Y')
                    y++;
                else if (ch == 'N')
                    n++;
                else if (ch == 'P')
                    p++;
                else if (ch == 'A')
                    a++;
                else 
                    System.out.println("ERROR"); // judge check
            }
            if (a >= y+n+p)
                System.out.println("need quorum"); 
            else if (y > n)
                System.out.println("yes"); 
            else if (y < n)
                System.out.println("no"); 
            else
                System.out.println("tie"); 
            s = in.nextLine();
        }
    }
}

File: _static/archives/2010/mcpc2010/solDir/voting.cpp
/* Voting, MCPC 2010 Problem D, C++ solution by Michael Goldwasser */
#include <iostream>
#include <fstream>
using namespace std;

fstream fin("voting.in");

int main() {

  while (true) {
    string votes;
    fin >> votes;
    if (votes == "#") break; // end of input

    int y,n,a,p;
    y = n = a = p = 0;
    for (int k=0; k < votes.size(); k++) {
      switch (votes[k]) {
      case 'Y':
        y++;
        break;
      case 'N':
        n++;
        break;
      case 'A':
        a++;
        break;
      case 'P':
        p++;
        break;
      default:
        cout << "OOPS" << endl;
      }
    }

    if (a >= y+n+p)
      cout << "need quorum" << endl;
    else if (y > n)
      cout << "yes" << endl;
    else if (n > y)
      cout << "no" << endl;
    else
      cout << "tie" << endl;

  }

}
